client/fingerprint/fingerprint.go

```diff
@@ -3,14 +3,27 @@ package fingerprint
 import (
 	"fmt"
 	"log"
+	"sort"
 	"time"
 
 	"github.com/hashicorp/nomad/client/config"
 	"github.com/hashicorp/nomad/nomad/structs"
 )
 
 // EmptyDuration is to be used by fingerprinters that are not periodic.
-const EmptyDuration = time.Duration(0)
+const (
+	EmptyDuration = time.Duration(0)
+)
+
+func init() {
+	// Initialize the list of available fingerprinters per platform.  Each
+	// platform defines its own list of available fingerprinters.
+	initFingerprints(builtinFingerprintMap)
```

Comment:
Heh, I started with that and talked myself out of it thinking it would be easier to just maintain the list by humans as a static list per platform given the infrequency at which this changes.  Pushed a fix that does exactly this.